modules/core/04-channel/keeper/grpc_query.go

```diff
@@ -555,25 +555,21 @@ func (k Keeper) NextSequenceSend(c context.Context, req *types.QueryNextSequence
 
 	ctx := sdk.UnwrapSDKContext(c)
 
-	channel, found := k.GetChannel(ctx, req.PortId, req.ChannelId)
+	_, found := k.GetChannel(ctx, req.PortId, req.ChannelId)
 	if !found {
 		return nil, status.Error(
 			codes.NotFound,
 			errorsmod.Wrapf(types.ErrChannelNotFound, "port-id: %s, channel-id %s", req.PortId, req.ChannelId).Error(),
 		)
 	}
```

Comment:
I think this block of code can be removed. There is no need to check the channel exists here if the channel itself is unused.

Reply:
i retained old code for conaistency of errors and ease of debug.

Reply:
i guess getchannel method at some point could be redesigned.

Reply:
Thank you, appreciate the attention and care for retaining the previous behaviour wrt errors etc. 

I would still recommend to remove the channel lookup as its unnecessary. If the channel does not exist then likewise, the `nextSequenceSend` will not be set and an error will be returned to the client.